crates/shared/src/sources/balancer_v2/pool_fetching.rs

```diff
@@ -156,30 +156,65 @@ pub enum BalancerFactoryKind {
     NoProtocolFeeLiquidityBootstrapping,
 }
 
+impl BalancerFactoryKind {
+    /// Returns a vector with supported factories for the specified chain ID.
+    ///
+    /// # Panics
+    ///
+    /// This associated function panics for unsupported chain IDs.
+    pub fn for_chain(chain_id: u64) -> Vec<Self> {
```

Comment:
How do you feel about panic vs empty Vec? If we say we know what networks are supported then empty might make more sense. No strong feeling on my side.

Reply:
Thinking about it a bit, we already fail in `BalancerContracts::new...` no Vault deployment (which implies no factories). So returning an empty list just makes it so we reuse that error which is easier to understand.

Will change.